hera_cal/nucal.py

```diff
@@ -0,0 +1,473 @@
+        ants=None,
+        ex_ants=None,
+        pols=None,
+        ex_pols=None,
+        min_bl_cut=None,
+        max_bl_cut=None,
+    ):
+        """
+        Filter radially redundant group to include/exclude the specified bls, antennas. and polarizations.
+        Arguments are evaluated, in order of increasing precedence: (pols, ex_pols, bls, ex_bls, ants, ex_ants,
+        min_bl_cut, max_bl_cut).
+
+        Parameters:
+        ----------
+        bls : list of tuples, default=None
+            baselines to include. Baselines of the form (i,j,pol) include that specific
+            visibility.  Baselines of the form (i,j) are broadcast across all polarizations present in reds.
+        ex_bls : list of tuples, default=None
+            same as bls, but excludes baselines.
+        ants : list of tuples, default=None
+            antennas to include. Only baselines where both antenna indices are in ants
+            are included.  Antennas of the form (i,pol) include that antenna/pol. Antennas of the form i are
+            broadcast across all polarizations present in reds.
+        ex_ants : list of tuples, default=None
+            same as ants, but excludes antennas.
+        pols : list of strings
+            polarizations to include in reds. e.g. ['nn','ee','ne','en']
+        ex_pols : list of strings
+            same as pols, but excludes polarizations.
+        min_bl_cut:
+            Cut baselines in the radially redundant group with lengths less than min_bl_cut
+        max_bl_cut:
+            Cut baselines in the radially redundant group with lengths less than min_bl_cut
+        """
+        _baselines = redcal.filter_reds(
+            [self._baselines],
+            bls=bls,
+            ex_bls=ex_bls,
+            ants=ants,
+            ex_ants=ex_ants,
+            pols=pols,
+            ex_pols=ex_pols,
+        )
+        if len(_baselines) == 0:
+            self._baselines = []
+            self.baseline_lengths = []
+        else:
+            new_bls = []
+            new_bls_lengths = []
+            for bls in _baselines[0]:
+                index = self._baselines.index(bls)
+                if min_bl_cut is not None and self.baseline_lengths[index] < min_bl_cut:
+                    continue
+                if max_bl_cut is not None and self.baseline_lengths[index] > max_bl_cut:
+                    continue
+                new_bls.append(bls)
+                new_bls_lengths.append(self.baseline_lengths[index])
+
+            self._baselines = new_bls
+            self.baseline_lengths = new_bls_lengths
+
+    def __iter__(self):
+        """Iterate through baselines in the radially redundant group"""
+        return iter(self._baselines)
+
+    def __len__(self):
+        """Return the length of the baselines list"""
+        return len(self._baselines)
+
+    def __getitem__(self, index):
+        """Get the baseline at the chosen index"""
+        return self._baselines[index]
+
```

Comment:
I'd suggest supporting __setitem__, though there may be other list methods you also want to support (see https://www.w3schools.com/python/python_ref_list.asp)